runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java

```diff
@@ -73,9 +73,9 @@ public void serialize(ByteBuf b) {
     }
 
     @Override
-    public void setEntry(ILogData entry) {
-        super.setEntry(entry);
-        this.getUpdates().forEach(x -> x.setEntry(entry));
+    public void setGlobalAddress(long address) {
+        super.setGlobalAddress(address);
```

Comment:
Is it a big change to make globalAddress a final ?

Reply:
I guess so. Not sure if we can do this because by the time a log entry is constructed, the global address is unknown.